ext/url/00_url.js

```diff
@@ -44,41 +44,37 @@
 
   // Helper functions
   function opUrlReparse(href, setter, value) {
-    return _urlParts(
-      ops.op_url_reparse(href, [setter, value]),
+    const status = ops.op_url_reparse(
+      href,
+      setter,
+      value,
+      componentsBuf.buffer,
     );
+    return getSerialization(status, href);
   }
+
   function opUrlParse(href, maybeBase) {
-    return _urlParts(ops.op_url_parse(href, maybeBase));
+    let status;
+    if (maybeBase === undefined) {
+      status = ops.op_url_parse(href, componentsBuf.buffer);
+    } else {
+      status = core.ops.op_url_parse_with_base(
+        href,
+        maybeBase,
+        componentsBuf.buffer,
+      );
+    }
+    return getSerialization(status, href);
   }
-  function _urlParts(internalParts) {
-    // WARNING: must match UrlParts serialization rust's url_result()
-    const {
-      0: href,
-      1: hash,
-      2: host,
-      3: hostname,
-      4: origin,
-      5: password,
-      6: pathname,
-      7: port,
-      8: protocol,
-      9: search,
-      10: username,
-    } = StringPrototypeSplit(internalParts, "\n");
-    return {
-      href,
-      hash,
-      host,
-      hostname,
-      origin,
-      password,
-      pathname,
-      port,
-      protocol,
-      search,
-      username,
-    };
+
+  function getSerialization(status, href) {
+    if (status === 0) {
+      return href;
+    } else if (status === 1) {
+      return core.ops.op_url_get_serialization();
+    } else {
+      throw new TypeError("Invalid URL");
+    }
   }
 
   class URLSearchParams {
```

Comment:
i did some benchmarking and there is a small change we can make here that passes all the wpt tests and bumps perf by 15-20% when api consumer accesses searchParams getter on URL and there are no query params, which is likely the common case. i can do this in a separate PR if you like as i think there are a few other small wins we can find here.

![Screenshot from 2022-09-09 23-31-31](https://user-images.githubusercontent.com/159073/189454701-36d365d1-f237-4e62-89b8-a8ccd0b333dd.png)

Reply:
actually, we could just move that up to the top of the function i think?

Reply:
Nice find! 

> i can do this in a separate PR 

yup, that'll be great